CRM/Utils/Check/Component/Cleanurls.php

```diff
@@ -0,0 +1,104 @@
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Utils_Check_Component_Cleanurls extends CRM_Utils_Check_Component {
+
+  /**
+   * For sites running in WordPress, make sure clean URLs are properly set in settings file.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public static function checkWpCleanurls() {
+    $config = CRM_Core_Config::singleton();
+    //Right now this config issue largely affects Wordpress only. This check could be applied to other CMS' but for now it's just WP
+    if ($config->userFramework != 'WordPress') {
+
+      return [];
+    }
+    $clean = $civicrm_setting['CIVICRM_CLEANURL'];
+    if ($clean == 1) {
+      //cleanURLs are enabled in CiviCRM, let's make sure the wordpress permalink settings and cache are actually correct by checking the first active contribution page
+      $contributionPages = \Civi\Api4\ContributionPage::get(FALSE)
+        ->addSelect('id')
+        ->addWhere('is_active', '=', TRUE)
+        ->setLimit(1)
+        ->execute();
+      if (count($contributionPages) > 0) {
+        $activePageId = $contributionPages[0]['id'];
+        $message = self::checkCleanPage('/contribute/transact/?reset=1&id=', $activePageId, $config);
+
+        return $message;
+      }
+      else {
+        //no active contribution pages, we can check an event page. This probably won't ever happen.
+        $eventPages = \Civi\Api4\Event::get(FALSE)
+          ->addSelect('id')
+          ->addWhere('is_active', '=', TRUE)
+          ->setLimit(1)
+          ->execute();
+        if (count($eventPages) > 0) {
+          $activePageId = $eventPages[0]['id'];
+          $message = self::checkCleanPage('/event/info/?reset=1&id=', $activePageId, $config);
+
+          return $message;
+        }
+        else {
+          //If there are no active event or contribution pages, we'll skip this check for now.
+
+          return [];
+        }
+      }
+    }
+    else {
+      //cleanURLs aren't enabled or aren't defined correctly in CiviCRM, admin should check civicrm.settings.php
+      $warning = 'Clean URLs are not enabled correctly in CiviCRM. This can lead to "valid id" errors for users registering for events or making donations. Check civicrm.settings.php and review <a href="https://docs.civicrm.org/sysadmin/en/latest/integration/wordpress/clean-urls/">the documentation</a> for more information.';
+
+      return [
+        new CRM_Utils_Check_Message(
+          __FUNCTION__,
+          $warning,
+          ts('Clean URLs Not Enabled'),
+          \Psr\Log\LogLevel::WARNING,
+          'fa-wordpress'
+        ),
+      ];
+    }
+  }
+
+  private static function checkCleanPage($slug, $id, $config) {
+    $client = new GuzzleHttp\Client();
+    $res = $client->head($config->userFrameworkBaseURL . $config->wpBasePage . $slug . $id);
+    $httpCode = $res->getStatusCode();
```

Comment:
There might be some situations where guzzle throws an exception. Can wrap this in a try/catch just so that system checks don't fail in that case.

Reply:
Ah it needs a backslash in front of GuzzleHttp, or add a `use` statement at the top of the file and then just put `Client` here.

Reply:
Okay cool i'll try that when I get back to this later. I thought it was set properly because I can see it hitting the url in my access log but the system page then fails to load so obviously something was wrong.